packages/flutter/test/material/bottom_navigation_bar_test.dart

```diff
@@ -851,6 +851,208 @@ void main() {
           ])));
     }, throwsA(isInstanceOf<AssertionError>()));
   });
+
+  testWidgets('BottomNavigationBar [showLabel]=false for shifting navbar, expect that there is no rendered text',
+      (WidgetTester tester) async {
+    final Widget widget = MaterialApp(
+      home: StatefulBuilder(
+        builder: (BuildContext context, StateSetter setState) {
+          return Scaffold(
+            bottomNavigationBar: BottomNavigationBar(
+              showLabel: false,
+              type: BottomNavigationBarType.shifting,
+              items: const <BottomNavigationBarItem>[
+                BottomNavigationBarItem(
+                  title: Text('Red'),
+                  backgroundColor: Colors.red,
+                  icon: Icon(Icons.dashboard),
+                ),
+                BottomNavigationBarItem(
+                  title: Text('Green'),
+                  backgroundColor: Colors.green,
+                  icon: Icon(Icons.menu),
+                ),
+              ],
+            ),
+          );
+        },
+      ),
+    );
+    await tester.pumpWidget(widget);
+    
+    expect(find.text('Red'), findsOneWidget);
+    expect(find.text('Green'), findsOneWidget);
+ 
+    final Finder navigationBarItems = find.descendant(
+      of: find.byType(BottomNavigationBar),
+      matching: find.byWidgetPredicate((Widget w) {
+        if (w is Opacity)
+          return w.opacity == 0.0;
+        return false;
+      }),
+    );
+
+    expect(tester.widget<Opacity>(navigationBarItems.first).opacity, 0.0);
+    expect(tester.widget<Opacity>(navigationBarItems.last).opacity, 0.0);
+  });
+
+  testWidgets('BottomNavigationBar [showLabel]=false for fixed navbar, expect that there is no rendered text',
+      (WidgetTester tester) async {
+    await tester.pumpWidget(
+      MaterialApp(
+        home: StatefulBuilder(
+          builder: (BuildContext context, StateSetter setState) {
+            return Scaffold(
+              bottomNavigationBar: BottomNavigationBar(
+                showLabel: false,
+                type: BottomNavigationBarType.fixed,
+                items: const <BottomNavigationBarItem>[
+                  BottomNavigationBarItem(
+                    title: Text('Red'),
+                    backgroundColor: Colors.red,
+                    icon: Icon(Icons.dashboard),
+                  ),
+                  BottomNavigationBarItem(
+                    title: Text('Green'),
+                    backgroundColor: Colors.green,
+                    icon: Icon(Icons.menu),
+                  ),
+                ],
+              ),
+            );
+          },
+        ),
+      ),
+    );
+    
+    expect(find.text('Red'), findsOneWidget);
+    expect(find.text('Green'), findsOneWidget);
+
+    final Finder navigationBarItems = find.descendant(
+      of: find.byType(BottomNavigationBar),
+      matching: find.byWidgetPredicate((Widget w) {
```

Comment:
Do you need to filter down with the predicate too or can use use `byType(Opacity)` here?

Reply:
I've removed filtering.